install.sh

```diff
@@ -606,6 +652,10 @@ ohai "Downloading and installing Homebrew..."
 (
   cd "${HOMEBREW_REPOSITORY}" >/dev/null || return
 
+  if [[ "$HOMEBREW_BREW_DEFAULT_GIT_REMOTE" != "$HOMEBREW_BREW_GIT_REMOTE" ]]; then
+    echo "HOMEBREW_BREW_GIT_REMOTE set: using $HOMEBREW_BREW_GIT_REMOTE for Homebrew/brew Git remote URL."
```

Comment:
👏🏻 for printing this. Another option to consider would be we prompt if this is non-default and otherwise use environment variables?

Reply:
This will only prompt the user that he or she is using a non-default URL, nothing more.

The variables from environment has been cleared at line 98:
```bash
unset HOMEBREW_{BREW,CORE}_GIT_REMOTE  # unset variables from environment
```

`HOMEBREW_{BREW,CORE}_GIT_REMOTE` is non-default, only when the user passes command line options to the script.

Reply:
Yeh, my thinking is we could use environment variables (your original idea) but stop the installation and prompt if they are set to non-default values? Thoughts?

Reply:
I prefer to make everything automated once the sudo privilege is checked. The installation process will be interrupted for non-default remotes if we prompt and wait. There will be three different locations that interrupt the process and wait for user input. One for sudo privilege check at the beginning, and two for remote URL confirmation in the intermediate.

Reply:
> I prefer to make everything automated once the sudo privilege is checked.

Given it's already not automated: we could make it a single prompt. I guess you'll need those variables set for Homebrew/brew anyway so as long as we notify and prompt it seems preferable to having to add e.g. argument parsing into this script.

Reply:
I put the prompts for Git remotes before `wait_for_user` now. We can read the remote URLs from environment variables, but it needs three commands to install:
```bash
export HOMEBREW_BREW_GIT_REMOTE="..."
export HOMEBREW_CORE_GIT_REMOTE="..."
bash -c "$(cat install.sh)"
```
It only need one command to install if the URLs are parsed from arguments:
```bash
bash -c "$(cat install.sh)" -b "..." -c "..."
```
So I think _parse the remotes from arguments_ is better.

On my devices, I am using non-default remotes for brew's repos, but I do not set `HOMEBREW_{BREW,CORE}_GIT_REMOTE` in my shell environment. `brew update` set the remote of local repos if `HOMEBREW_{BREW,CORE}_GIT_REMOTE` is set to non-default value. However, `brew update` does not change the remote of local repos when those env-vars are not set, even if `git -C "$(brew --repo)" remote get-url origin` reports a non-default URL. In other words, users only need to set these env-vars at the begining of installation. Once the installation is complete, these variables can be unset. Users do not need them for daily use.

Reply:
> It only need one command to install if the URLs are parsed from arguments:

You'll need to set those variables after installation, though, or `brew doctor` will complain, which isn't something we want to be possible with something from the official installation script.

Reply:
Oh I see. I have never ran `brew doctor` before, so I didn't know that. So users are recommanded to set `HOMEBREW_{BREW,CORE}_GIT_REMOTE` if they are using non-default ones, is that correct? I revert to the original approach, _read Homebrew remote URLs from environment variables and prompt for non-default values_.